### ISSUE TITLE
helm3 - --timeout flag ignored

### ISSUE THREAD

**Issue by novoxudonoser** [helm/helm#7210]
Output of `helm version`:
```
version.BuildInfo{Version:"v3.0", GitCommit:"", GitTreeState:"", GoVersion:"go1.12.9"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.9", GitCommit:"3e4f6a92de5f259ef313ad876bb008897f6a98f0", GitTreeState:"archive", BuildDate:"1970-01-01T00:00:01Z", GoVersion:"go1.11.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
Minikube, self-hosted Kubernetes

I have such job:
```
apiVersion: batch/v1
kind: Job
metadata:
  name: db-migrate
  annotations:
    "helm.sh/hook": pre-install,pre-upgrade
    "helm.sh/hook-delete-policy": hook-succeeded
spec:
  template:
    metadata:
      name: db-migrate
    spec:
      restartPolicy: Never
      containers:
        - name: migrade-exmple
          image: alpine:latest
          command: ["sleep", "300"]

```

I install chart with:
```
helm upgrade --install myname . --timeout 3000s
```

After 60s no matter what in the timeout flag I get:
```
Release "myname" does not exist. Installing it now.
Error: failed pre-install: watch closed before UntilWithoutRetry timeout
```

**Comment by novoxudonoser** [comment]
@bacongobbler can you help with reviewing this issue?

**Comment by bacongobbler** [comment]
have you looked at #2025? Most issues related to timeouts closing early are due to some external factor. In most cases it's the load balancer fronting the Kubernetes API server closing the connection.

Let us know if you find something.

**Comment by novoxudonoser** [comment]
@bacongobbler had locked in #2025 it is not it. First I am running this installation on local minikube what and there cant be any issues with the network (during all my interaction with it there weren't any). 

Second, my error message is `Error: failed pre-install: watch closed before UntilWithoutRetry timeout` what says about pre-install hook timeout. And more - it happens directly after 60s what is the default timeout, that why I think that timeout flag is ignored.

**Comment by hickeyma** [comment]
@novoxudonoser This looks like it might be the same issue as #2918. It looks like that fix delivered in PR #6014 for Helm 2 was not ported to Helm 3. I have pushed #7217 for Helm 3 fix.

**Comment by novoxudonoser** [comment]
The #7217 fixes this issue. @bacongobbler you were right the issue was about TCP timeout of the load balancer, but still, it was a bug cause no "heartbeats" while waiting for hook where send.
